clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py

```diff
@@ -26,82 +33,121 @@ class NegativeControlWasPositive(AnalysisServiceException):
     pass
 
 
+class FailedControl(AnalysisServiceException):
+    pass
+
+
 class RTPCRAnalysisService(object):
+    """
+    Create a new RTPCRAnalysis Service:
+        RTPCRAnalysisService(covid_reporter_key="FAM",
```

Comment:
Would it perhaps be a tiny bit clearer to just send in `FAM-CT` and `HEX-CT` rather than appending that in the logic?

Reply:
Yes. I do think that would make sense, especially taken together with your suggestion of creating sub-classes for different instruments.